admin/release.py

```diff
@@ -406,7 +406,12 @@ def publish_homebrew_recipe(homebrew_repo_url, version, source_bucket,
 
     homebrew_repo.index.add([recipe])
     homebrew_repo.index.commit('Add recipe for Flocker version ' + version)
+
+    # Sometimes this raises an index error, and it seems to be a race
+    # condition. There should probably be a loop until push succeeds or
+    # whatever condition is necessary for it to succeed is met. FLOC-2043.
```

Comment:
More useful to display the return value of `push` if this error occurs., particularly as GitPython docs are vague as to actual return value ("If the operation fails completely, the length of the returned IterableList will be null."):
```py
push_info = homebrew_repo.remotes.origin.push(homebrew_repo.head)
assert hasattr(push_info, '__len__') and len(push_info) == 1, push_info
push_info = push_info[0]
```